include/aspect/material_model/interface.h

```diff
@@ -317,6 +321,28 @@ namespace aspect
          */
         typename DoFHandler<dim>::active_cell_iterator current_cell;
 
+        /**
+         * Vector of shared pointers to additional material model input
+         * objects that can be added to MaterialModelInputs. By default,
+         * no inputs are added.
+         */
+        std::vector<std_cxx11::shared_ptr<AdditionalMaterialInputs<dim> > > additional_inputs;
+
+        /**
+         * Given an additional material model input class as explicitly specified
+         * template argument, returns a pointer to this additional material model
+         * input object if it is used in the current simulation.
+         * If the output does not exist, a null pointer is returned.
+         */
+        template <class AdditionalInputType>
```

Comment:
I would like this to have the same structure as the `has_matching_plugin` `get_matching_plugin` functions in all the other plugin systems. This avoids using null pointers. But since this is identical to the material model outputs, I would say let's create an issue about it, and keep it like this for this PR.